src/hooks/use-click-away.ts

```diff
@@ -28,7 +28,7 @@ export function useClickAway(
     const handleAwayClick = (event: Event) => {
       if (
         container.current &&
-        !container.current.contains(event.target as Node)
+        !event.composedPath().includes(container.current)
```

Comment:
This definitely deserves a comment to explain why we're not using the more obvious `container.current.contains(event.target)` test. Something like:

```js
// We test the composed path here to handle the case where the clicked element is
// removed from the DOM (eg. by a re-render in a child component) before this callback is run.
// The composed path reflects the DOM hierarchy at the time the event was dispatched.
```